src/main/io/gps.c

```diff
@@ -494,6 +494,15 @@ void gpsUpdate(timeUs_t currentTimeUs)
             gpsNewData(serialRead(gpsPort));
     }
 
+    // GPS data received via MSP
+    if (GPS_update & 2) {
+        gpsSetState(GPS_RECEIVING_DATA);
```

Comment:
(this may cause problems when both MSP and GPS is connected, but that's probably acceptable ... )

Reply:
A solution could be to `} else if (GPS_update & 2) {` attached to the previous `if` statement. Then it ignores any data sent via MSP if a GPS is connected directly.